File: C#/GrandCircus/AnimalBase.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Xml.Schema;

namespace RemoteLearning.GrandCircus
{
    internal class   AnimalBase : IAnimal
    {

      public string SpeciesName { get; set; }
      public string Name {  get; set; }

       public AnimalBase (string name, string speciesName)
        {
            Name = name;
            SpeciesName = speciesName;
        }
        public  virtual string MakeSound()
        {
            return "Sound made by the animal: ";
        }
    }
}

File: C++/GrandCircusC++/GrandCircusC++/AnimalBase.cpp
#include "AnimalBase.h"

AnimalBase::AnimalBase(string SpeciesName, string Name)
    : SpeciesName(SpeciesName), Name(Name) {
}

string AnimalBase::getName() { return Name; }
string AnimalBase::getSpeciesName() { return SpeciesName; }
string AnimalBase::MakeSound() { return "Sound made by Animal: "; }